lib/ansible/galaxy/dependency_resolution/providers.py

```diff
@@ -182,7 +188,7 @@ def get_preference(
             return float('-inf')
         return len(candidates)
 
-    def find_matches(self, requirements):
+    def find_matches(self, identifier, requirements, incompatibilities):
```

Comment:
The new implementation should support signatures for a number of resolvelib versions. They could be defined dynamically in runtime.

Reply:
Refer to my above feedback: if ansible 2.12.0 is target for Python 3.8+ (which brake most legacy 2.5+ years ago distributions), keeping a multiple support for resolvelib 0.6.0/0.7.0/0.8.0/etc are most likely pointless?

Reply:
It is a reason to keep the API compatible.